record-editor.py

```diff
@@ -1691,40 +1721,42 @@ def create_layout( self ):
         editing_layout.addWidget( self.artDateLineEdit,
                                   5, 1 )
 
-
-        editing_box = QGroupBox()
-        editing_box.setLayout( editing_layout )
-
         #   vertical layout of the photo preview and everything else.
         main_layout = QVBoxLayout()
         main_layout.setContentsMargins( 0, 0, 0, 0 )
         main_layout.setSpacing( 0 )
 
         main_layout.addWidget( self.photoPreview )
         main_layout.setAlignment( self.photoPreview, Qt.AlignCenter ) # the preview should be centered.
-        main_layout.addWidget( editing_box )
+        main_layout.addLayout( editing_layout )
+        main_layout.setStretchFactor( self.photoPreview, 1 )
 
         self.centralWidget = QGroupBox()
         self.centralWidget.setLayout( main_layout )
 
+        self.setCentralWidget( self.centralWidget )
+
     def create_menus( self ):
         """
-        Creats the menus for an ArtRecordEditor.
+        Creates the menus for an ArtRecordEditor.
 
         Takes no arguments.
 
         Returns nothing.
         """
-
-        self.closeAct  = QAction( "C&lose", self, shortcut="Ctrl+W",
-                                  triggered=self.close )
+        # XXX: for whatever reason, closing the window via menu action
+        # when there are more than two QMainWindow instances causes
+        # a hard crash. For now the user will have to resort to using
+        # the X button, which does not cause the crash.
+        # self.closeAct  = QAction( "C&lose", self, shortcut="Ctrl+W",
+        #                           triggered=self.close )
```

Comment:
Not sure where this crash comes from.  On an Ubuntu 14.04.4 LTS system, a Fedora 20, and a Fedora 24 system the keyboard shortcut works without issue and has for some 4500 record entries (~4000 on the Ubuntu system, ~500 on the Fedora systems).  Is it present in the original code before the other changes in this commit?